wandb/sdk/wandb_init.py

```diff
@@ -1560,7 +1560,21 @@ def init(  # noqa: C901
             if run_settings.x_server_side_derived_summary:
                 init_telemetry.feature.server_side_derived_summary = True
 
-            return wi.init(run_settings, run_config, run_printer)
+            run = wi.init(run_settings, run_config, run_printer)
+
+            # Set up automatic Weave integration if Weave is installed
```

Comment:
I know this is not the typical place we install integrations.  I'm adding it here because we specifically want to init `weave` when we init `wandb`.